composer/callbacks/checkpoint_saver.py

```diff
@@ -297,29 +271,44 @@ def __init__(
         self.save_interval = save_interval
         self.last_checkpoint_batch: Optional[Time] = None
 
-        self.folder = folder
-
-        self.filename = PartialFilePath(filename.lstrip('/'), folder)
-        self.latest_filename = PartialFilePath(latest_filename.lstrip('/'), folder) if latest_filename else None
-
-        self.artifact_name = PartialFilePath(artifact_name) if artifact_name else None
-        self.latest_artifact_name = PartialFilePath(latest_artifact_name) if latest_artifact_name else None
+        
+        self.folder = urlparse(folder).path
```

Comment:
no need to duplicate the urlparse call here, can make a variable `parsed_url`

Reply:
fixed